contracts/AccessControl/RoleBasedAccessControl.sol

```diff
@@ -0,0 +1,147 @@
+pragma solidity ^0.4.11;
+
+import './IAccessPolicy.sol';
+import './IAccessControlled.sol';
+import './AccessControlled.sol';
+
+contract RoleBasedAccessControl is IAccessPolicy, AccessControlled {
+
+    ////////////////
+    // Types
+    ////////////////
+
+    // Łukasiewicz logic values
+    enum TriState {
+        Unset,
+        True,
+        False
+    }
+
+    ////////////////
+    // Constants
+    ////////////////
+
+    IAccessControlled public constant GLOBAL = IAccessControlled(0x0);
+
+    ////////////////
+    // State
+    ////////////////
+
+    mapping (address => mapping(bytes32 => mapping(address => TriState))) access;
```

Comment:
Not too much more complicated. I added a query `getUsers(object, role)` that returns all users that have a permission set (either allow or deny).